File: calculator.py
def calculate(num1, num2, operation):
    if operation == '+':
        return num1 + num2
    elif operation == '-':
        return num1 - num2
    elif operation == '*':
        return num1 * num2
    elif operation == '/':
        if num2 != 0:
            return num1 / num2
        else:
            raise ValueError("Error: Division by zero")
    else:
        raise ValueError("Error: Unsupported operation")

def main():
    print("Simple Calculator")

    while True:
        try:
            num1 = float(input("Enter first number: "))
            operation = input("Enter operation (+, -, *, /): ")
            num2 = float(input("Enter second number: "))

            result = calculate(num1, num2, operation)
            print(f"Result: {result}")

            continue_calc = input("Do you want to perform another calculation? (yes/no): ")
            if continue_calc.lower() != 'yes':
                break
        except ValueError as e:
            print(e)
        except Exception as e:
            print("An unexpected error occurred:", e)

    print("Goodbye!")

if __name__ == "__main__":
    main()

File: calculator.cs
using System;

class Calculator
{
    // Method to perform calculations
    static double Calculate(double num1, double num2, string operation)
    {
        switch (operation)
        {
            case "+":
                return num1 + num2;
            case "-":
                return num1 - num2;
            case "*":
                return num1 * num2;
            case "/":
                if (num2 != 0)
                {
                    return num1 / num2;
                }
                else
                {
                    throw new DivideByZeroException("Error: Division by zero");
                }
            default:
                throw new InvalidOperationException("Error: Unsupported operation");
        }
    }

    static void Main(string[] args)
    {
        Console.WriteLine("Simple Calculator");

        while (true)
        {
            try
            {
                Console.Write("Enter first number: ");
                double num1 = Convert.ToDouble(Console.ReadLine());

                Console.Write("Enter operation (+, -, *, /): ");
                string operation = Console.ReadLine();

                Console.Write("Enter second number: ");
                double num2 = Convert.ToDouble(Console.ReadLine());

                double result = Calculate(num1, num2, operation);
                Console.WriteLine($"Result: {result}");

                Console.Write("Do you want to perform another calculation? (yes/no): ");
                string continueCalc = Console.ReadLine();
                if (continueCalc.ToLower() != "yes")
                {
                    break;
                }
            }
            catch (FormatException)
            {
                Console.WriteLine("Invalid input. Please enter numeric values.");
            }
            catch (DivideByZeroException ex)
            {
                Console.WriteLine(ex.Message);
            }
            catch (InvalidOperationException ex)
            {
                Console.WriteLine(ex.Message);
            }
            catch (Exception ex)
            {
                Console.WriteLine("An unexpected error occurred: " + ex.Message);
            }
        }

        Console.WriteLine("Goodbye!");
    }
}
